uber/site_sections/registration.py

```diff
@@ -103,7 +104,7 @@ def form(self, session, message='', return_to='', omit_badge='', check_in='', **
             message = check_everything(attendee)
             if not message:
                 # Free group badges are only considered 'registered' when they are actually claimed.
-                if attendee.paid == c.PAID_BY_GROUP and attendee.group.cost == 0:
+                if attendee.paid == c.PAID_BY_GROUP and attendee.group_id and attendee.group.cost == 0:
```

Comment:
If someone fills out the on-site reg form and says they're claiming a group badge, they'll be marked as ``c.PAID_BY_GROUP`` without actually belonging to a group.  Therefore, we need to check that ``attendee.group_id`` exists before we check the group cost.